src/proto/single/resp.rs

```diff
@@ -118,6 +120,92 @@ pub async fn read_ok<R: AsyncBufRead + Unpin>(r: R) -> Result<(), Error> {
     Err(bad("server ok", &rr).into())
 }
 
+// ----------------------------------------------
+
+/// Faktory service stats.
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[non_exhaustive]
+pub struct DataSnapshot {
+    /// Total number of job failures.
+    pub total_failures: u64,
+
+    /// Total number of processed jobs.
+    pub total_processed: u64,
+
+    /// Total number of enqueued jobs.
+    pub total_enqueued: u64,
+
+    /// Total number of queues.
+    pub total_queues: u64,
+
+    /// Queues stats.
+    ///
+    /// A mapping between a queue name and its size (number of jobs on the queue).
+    /// The keys of this map effectively make up a list of queues that are currently
+    /// registered in the Faktory service.
+    pub queues: HashMap<String, u64>,
+
+    /// ***Deprecated***. Faktory's task runner stats.
+    ///
+    /// Note that this is exposed as a "generic" `serde_json::Value` since this info
+    /// belongs to deep implementation details of the Faktory service.
+    #[deprecated]
+    pub tasks: serde_json::Value,
+}
+
+/// Faktory's server process stats.
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct ServerSnapshot {
+    /// Faktory's description (e.g. "Faktory").
```

Comment:
```suggestion
    /// Description of the server process (e.g. "Faktory").
```

Reply:
When will this ever not be "Faktory"?

Reply:
I can easily imagine it be `Faktory Enterprise` or `Faktory Valkey`.

Reply:
EDIT: even though it is now simply ["Faktory"](https://github.com/rustworthy/faktory/blob/ec73667a31f31242fbbbab9dc5d2fd7cf74c3ce9/client/faktory.go#L4)